patches/build-android-gyp-lint.py.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/build/android/gyp/lint.py b/build/android/gyp/lint.py
+index f3d85f78ec1298c4e57e5811a3faede44edbc5f0..b7a0de5074e11681ee970c9b768c6e4b797585ee 100755
+--- a/build/android/gyp/lint.py
++++ b/build/android/gyp/lint.py
+@@ -237,6 +237,7 @@ def _RunLint(custom_lint_jar_path,
+   else:
+     creating_baseline = False
+     lint_xmx = '2G'
++    lint_xmx = '4G'
```

Comment:
I am not sure if it's worth it here since it's only a one-line change, but we do have a new way of overriding python scripts https://github.com/brave/brave-browser/wiki/Patching-Chromium#patching-py-files. You could probably use `override_utils.override_scope_variable` here, but I'll defer to @goodov on if it's worth it here.

Reply:
tbh I'd prefer to not introduce direct python patches for exact lines if it can be solved with a chromium_src-inlined patch. Most patches start as a "single-line" and then at some point something else creeps-in, so the patch is not a single-line anymore.

One approach to do the change you need is to override `build_utils.JavaCmd` with `override_utils.override_method` to always pass `4G` to `xms` parameter.


`override_utils.override_scope_variable` is used for other things, so it's not applicable here. It should actually be called `scoped_variable_override` 😅, because it's a temporary override that reverts the value after leaving `with ...:` scope. I might need to rename it to not confuse people.

Reply:
> thanks @goodov @mkarolin , fixed at [faf2e3c](https://github.com/brave/brave-core/commit/faf2e3c996282d0cd8e68273fa2abb79fcd89e38)
> 
> This approach causes each java process now have 4G xmx, but I didn't experience any trouble, as this parameter is for maximum heap size, and memory consumed only when required.

I think you can do this change only as a part of `lint.py` like so:
```
chromium_src/build/android/gyp/lint.py:

import override_utils

@override_utils.override_function(build_utils)
def JavaCmd(original_function, xmx='1G'):
    # Override to pass xmx='4G', to fix error
    # java.lang.OutOfMemoryError: Java heap space
    # for Android incremental builds

    return original_function('4G')

###

lint.py.patch:

diff --git a/build/android/gyp/lint.py b/build/android/gyp/lint.py
index f3d85f78ec129..3926e1c077e33 100755
--- a/build/android/gyp/lint.py
+++ b/build/android/gyp/lint.py
@@ -541,5 +541,6 @@ def main():
     action_helpers.write_depfile(args.depfile, args.stamp, depfile_deps)
 
 
+from brave_chromium_utils import inline_chromium_src_override; inline_chromium_src_override(globals(), locals())
 if __name__ == '__main__':
   sys.exit(main())
```

Reply:
thanks @goodov , I wasn't aware it can be overridden for the scope as in your example

right now xmx is set only for lint, no more worries remaining.

fixed at https://github.com/brave/brave-core/pull/22026/commits/f7e2cabeb4ed1c402ef74ebf1453eaeaf66fb1b7